### ISSUE TITLE
The validation set is the training set and there is no test loss

### ISSUE THREAD

**Issue by jzilly** [thuml/Transfer-Learning-Library#15]
Hello everyone, while trying to replicate some of your results using the DANN method on the Office31 dataset, we stumbled upon what appears like a missing validation and test set. 
At the moment when running DANN (https://github.com/thuml/Transfer-Learning-Library/blob/master/examples/dann.py), the validation set is the same as the training set. This leads to serious overfitting on the test set. 
I would appreciate if you can create a completely separate validation and test split (no overlapping images) of each subset of the Office31 dataset. This is very important because currently all results cannot be relied upon, since the model has access to the test images already during training. The current results are likely significantly better than they could logically be for such a small dataset, even when using pretraining.

**Comment by microhhh** [comment]
This is a historical problem. In DA related papers, the data set has always been used in this way, but the validation set cannot be used for training, only for evaluating, which is equivalent to looking at the performance on the test set in each epoch. If necessary, you can divide the data in txt by yourself.

**Comment by microhhh** [comment]
More, the data set is not divided, because the Office-31 dataset is too small, And if the test set is too small, it will not be able to accurately evaluate the performance. In DomainNet, the dataset has been divided.

**Comment by jzilly** [comment]
Ok I think I understand now. I looked a little through a few code bases and saw both versions (with and without separate test set). Thinking about I think both problem settings are valid in their own right. We as a community should give them different names however, for example "domain adaptation" (with test set) and "online domain adaptation" (test set = target train set). The "online" process (test set is allowed to adapt model such as in this code base) mirrors what people have been doing with RNNs when adapting them online on the test set. 
Thank you for the clarification.
